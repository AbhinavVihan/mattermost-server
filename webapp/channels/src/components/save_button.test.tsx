// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {screen} from '@testing-library/react';

import SaveButton from 'components/save_button';
import {renderWithIntl} from 'tests/react_testing_utils';

describe('components/SaveButton', () => {
    const baseProps = {
        saving: false,
    };

    test('should match snapshot, on defaultMessage', () => {
        const {container} = renderWithIntl(
            <SaveButton
                {...baseProps}
                defaultMessage='Go'
            />,
        );

        expect(screen.getAllByTestId('saveSetting')[0]).not.toBeDisabled();
        expect(container).toMatchSnapshot();
    });

    test('should match snapshot, on savingMessage', () => {
        const props = {...baseProps, saving: true, disabled: true};
        const {container} = renderWithIntl(
            <SaveButton
                {...props}
                savingMessage='Saving Config...'
            />,
        );

        expect(screen.getAllByTestId('saveSetting')[0]).toBeDisabled();
        expect(container).toMatchSnapshot();
    });

    test('should match snapshot, extraClasses', () => {
        const props = {...baseProps, extraClasses: 'some-class'};
        const {container} = renderWithIntl(
            <SaveButton {...props}/>,
        );

        expect(container).toMatchSnapshot();
    });
});
