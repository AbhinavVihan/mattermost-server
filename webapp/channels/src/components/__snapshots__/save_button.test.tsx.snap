// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SaveButton should match snapshot, extraClasses 1`] = `
<div>
  <button
    class="save-button btn btn-primary some-class"
    data-testid="saveSetting"
    id="saveSetting"
    type="submit"
  >
    <span>
      Save
    </span>
  </button>
</div>
`;

exports[`components/SaveButton should match snapshot, on defaultMessage 1`] = `
<div>
  <button
    class="save-button btn btn-primary"
    data-testid="saveSetting"
    id="saveSetting"
    type="submit"
  >
    <span>
      Go
    </span>
  </button>
</div>
`;

exports[`components/SaveButton should match snapshot, on savingMessage 1`] = `
<div>
  <button
    class="save-button btn btn-primary"
    data-testid="saveSetting"
    disabled=""
    id="saveSetting"
    type="submit"
  >
    <span
      class="LoadingSpinner with-text"
      id="loadingSpinner"
    >
      <span
        class="fa fa-spinner fa-fw fa-pulse spinner"
        title="Loading Icon"
      />
      Saving Config...
    </span>
  </button>
</div>
`;
